sys/kern/sysv_msg.c

```diff
@@ -450,16 +447,14 @@ msq_remove(struct msqid_kernel *msqkptr)
 	msqkptr->cred = NULL;
 
 	/* Free the message headers */
-	msghdr = msqkptr->first_msg;
-	while (msghdr != NULL) {
-		struct msg *msghdr_tmp;
+	while (!TAILQ_EMPTY(&msqkptr->queue)) {
```

Comment:
Alternatively instead of `TAILQ_EMPTY`/`TAILQ_FIRST` you could maybe use `TAILQ_FOREACH_SAFE`.  You need an extra `nmsghdr` variable declared, but these two lines would collapse down to just one.